src/charm.py

```diff
@@ -82,78 +86,108 @@ def __init__(self, *args):
         self._database = DatabaseRequires(
             self,
             relation_name=DATABASE_RELATION_NAME,
-            database_name=DATABASE_NAME,
+            database_name=DEFAULT_DATABASE_NAME,
+            extra_user_roles="admin",
+        )
+        self._auth_database = DatabaseRequires(
+            self,
+            relation_name="auth_database",
+            database_name=AUTH_DATABASE_NAME,
             extra_user_roles="admin",
         )
         self._sdcore_management = SdcoreManagementProvides(self, SDCORE_MANAGEMENT_RELATION_NAME)
         self.unit.set_ports(GRPC_PORT, WEBUI_URL_PORT)
-
-        self.framework.observe(self.on.update_status, self._configure)
-        self.framework.observe(self.on.webui_pebble_ready, self._configure)
-        self.framework.observe(self.on.database_relation_joined, self._configure)
+        self.framework.observe(self.on.webui_pebble_ready, self._configure_webui)
+        self.framework.observe(self.on.database_relation_joined, self._configure_webui)
         self.framework.observe(self.on.database_relation_broken, self._on_database_relation_broken)
-        self.framework.observe(self._database.on.database_created, self._configure_db)
-        self.framework.observe(self._database.on.endpoints_changed, self._configure_db)
+        self.framework.observe(self._database.on.database_created, self._configure_webui)
+        self.framework.observe(self._auth_database.on.database_created, self._configure_webui)
+        self.framework.observe(self._database.on.endpoints_changed, self._configure_webui)
+        self.framework.observe(self._auth_database.on.endpoints_changed, self._configure_webui)
         self.framework.observe(
             self.on.sdcore_management_relation_joined, self._publish_sdcore_management_url
         )
         # Handling config changed event to publish the new url if the unit reboots and gets new IP
         self.framework.observe(self.on.config_changed, self._publish_sdcore_management_url)
 
-    def _configure(self, event: EventBase) -> None:
-        """Juju event handler.
+    def _configure_webui(self, event: EventBase) -> None:
+        """Main callback function of the Webui operator.
 
-        Sets unit status, writes configuration file adds pebble layer.
+        Handles config changes.
+        Manages pebble layer and Juju unit status.
 
         Args:
-            event (EventBase): Juju event.
+            event: Juju event
         """
-        if not self._database_relation_is_created():
-            self.unit.status = BlockedStatus("Waiting for database relation to be created")
+        for relation in ["database", "auth_database"]:
+            if not self._relation_created(relation):
+                self.unit.status = BlockedStatus(f"Waiting for {relation} relation to be created")
+                return
+        if not self._database_is_available():
```

Comment:
I thought that this method included both databases so maybe we can find a better name

Reply:
This method only checks common database availability and the method name has been changed accordingly.